nixpkgs/nixpkgs.bzl

```diff
@@ -59,125 +59,25 @@ nixpkgs_local_repository = repository_rule(
-        fail("Specify one of 'nix_file' or 'nix_file_content', but not both.")
-    elif repository_ctx.attr.nix_file:
-        repository_ctx.symlink(repository_ctx.attr.nix_file, "default.nix")
-    elif repository_ctx.attr.nix_file_content:
-        expr_args = ["-E", repository_ctx.attr.nix_file_content]
-    elif not repositories:
-        fail(strFailureImplicitNixpkgs)
-    else:
-        expr_args = ["-E", "import <nixpkgs> {}"]
-
-    _symlink_nix_file_deps(repository_ctx, repository_ctx.attr.nix_file_deps)
-
-    expr_args.extend([
-        "-A",
-        repository_ctx.attr.attribute_path if repository_ctx.attr.nix_file or repository_ctx.attr.nix_file_content else repository_ctx.attr.attribute_path or repository_ctx.attr.name,
-        # Creating an out link prevents nix from garbage collecting the store path.
-        # nixpkgs uses `nix-support/` for such house-keeping files, so we mirror them
-        # and use `bazel-support/`, under the assumption that no nix package has
-        # a file named `bazel-support` in its root.
-        # A `bazel clean` deletes the symlink and thus nix is free to garbage collect
-        # the store path.
-        "--out-link",
-        "bazel-support/nix-out-link",
-    ])
-
-    expr_args.extend(repository_ctx.attr.nixopts)
-
-    # If repositories is not set, leave empty so nix will fail
-    # unless a pinned nixpkgs is set in the `nix_file` attribute.
-    nix_path = ""
-    if repositories:
-        nix_path = ":".join(
-            [
-                (path_name + "=" + str(repository_ctx.path(target)))
-                for (target, path_name) in repositories.items()
-            ],
-        )
-    elif not (repository_ctx.attr.nix_file or repository_ctx.attr.nix_file_content):
-        fail(strFailureImplicitNixpkgs)
-
-
-    if not_supported and fail_not_supported:
-        fail("Platform is not supported (see 'fail_not_supported')")
-    elif not_supported:
-        return
-    else:
-        nix_build_path = _executable_path(
-            repository_ctx,
-            "nix-build",
-            extra_msg = "See: https://nixos.org/nix/",
-        )
-        nix_build = [nix_build_path] + expr_args
-
-        # Large enough integer that Bazel can still parse. We don't have
-        # access to MAX_INT and 0 is not a valid timeout so this is as good
-        # as we can do.
-        timeout = 1073741824
-        exec_result = _execute_or_fail(
-            repository_ctx,
-            nix_build,
-            failure_message = "Cannot build Nix attribute '{}'.".format(
-                repository_ctx.attr.attribute_path,
+def nixpkgs_package(
+    name,
+    repositories = None,
+    repository = None,
+    nixopts = [],
+    **kwargs
+    ):
+  nixpkgs_packages(
+      name = name + "__drvfile",
+      repository = repository,
+      repositories = repositories,
+      nixopts = nixopts,
+      packages = {
+          name: dict(
+              nixops = nixopts,
+              **kwargs
```

Comment:
We can’t fail with both `nix_file` and `attribute_path` are supplied, that is a regression.

Instead, it should ask for the attribute in `nix_file`.